File: ios/Radio/RadioStation.swift
//
//  RadioStation.swift
//  Radio Lëtzebuerg
//
//  Model for radio stations - loads dynamically from JSON
//

import Foundation

struct RadioStation: Identifiable, Codable {
    let id: String
    let name: String
    let streamURL: String
    let logoImageName: String
    let websiteURL: String?
    let isEnabled: Bool?  // NEW: Si false, la radio est désactivée
    
    // Helper computed property
    var enabled: Bool {
        return isEnabled ?? true  // Par défaut: activée
    }
    
    // Computed property to get stations from loader
    static var stations: [RadioStation] {
        return RadioStationLoader.shared.stations
    }
}

File: android/app/src/main/java/com/florentin/letzlisten/data/RadioStation.kt
package com.florentin.letzlisten.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RadioStation(
    val id: String,
    val name: String,
    @SerialName("streamURL") val streamUrl: String,
    @SerialName("logoImageName") val logoImageName: String? = null,
    @SerialName("websiteURL") val websiteUrl: String? = null,
    @SerialName("isEnabled") val isEnabled: Boolean = true
)

@Serializable
data class StationsResponse(
    val version: String,
    val stations: List<RadioStation>
)
